packages/core/src/external/mpi/patient-incoming-schema.ts

```diff
@@ -0,0 +1,130 @@
+import { USState } from "../../domain/geographic-locations";
+import { Patient as FHIRPatient, ContactPoint } from "@medplum/fhirtypes";
+
+// TODO whole file should be migrated into mirth replacement module once we pass verification with testing partners.
+
+export type LivingSubjectId = {
+  extension?: string;
+  root?: string;
+};
+
+export type PrincipalCareProviderId = {
+  extension?: string;
+  root?: string;
+};
+
+export const generalTypes = ["passport", "ssn", "medicare"] as const;
+export const driversLicenseType = ["driversLicense"] as const;
+export type GeneralTypes = (typeof generalTypes)[number];
+export type DriverLicenseType = (typeof driversLicenseType)[number];
+
+export type Period =
+  | {
+      start: string;
+      end?: string;
+    }
+  | {
+      start?: string;
+      end: string;
+    };
+
+export type BaseIdentifier = {
+  period?: Period;
+  assigner?: string;
+};
+
+export type PersonalIdentifier = BaseIdentifier & {
+  type: DriverLicenseType;
+  value: string;
+  state: USState;
+};
+
+export type DriversLicense = {
+  value: string;
+  state: USState;
+};
+
+export const genderAtBirthTypes = ["F", "M"] as const;
+export type GenderAtBirth = (typeof genderAtBirthTypes)[number];
+
+export type Address = {
+  addressLine1: string;
+  addressLine2?: string;
+  city: string;
+  state: USState;
+  zip: string;
+  country?: string;
+};
+
+export type ContactTypes = "phone" | "fax" | "email" | "pager" | "url" | "sms" | "other";
+
+export type Contact = Partial<Record<ContactTypes, string | undefined>>;
+
+// modify address schema to have type of AddressLine2 be string? instead of string | undefined
+
+export type PatientDataMPI = {
```

Comment:
Do we absolutely need to create this type? Or, do we really want to make this as spread out through the app like we're doing it now?

Essentially, we have a `Patient` that was supposed to be our domain, or the main representation of a patient in our code, and this is introducing a new shape that's not compatible with the domain one.

Also, this is living on a `patient-incoming-schema`:
- schema is specific to external interfaces into the App (http and DB);
- incoming is specific to a context (CQ and incoming), so defs not something to be "core" to our domain.

Reply:
Well for starters, if the primary MPI operations are happening in core, we need a patient data type there. And `PatientData` has so much app specific logic that it would be a disaster to migrate over, plus a lot of it isn't relevant since it isn't demographic related. 
So creating a new type in core is required. The only really important field that PatientDataMPI adds is the id field. We could try to make the data types in core and api cross compatible so we dont need to do a conversion, but I am not sure if that is a thing given that api calls core and core cant inherit from api only vice versa. 

A second point about making it spread out across the app. As of right now, the only place in the app this is being used is in `getPatientsByDemo`. And its being converted from `PatientData` to `PatientDataMPI` and back again there, so its use is pretty encapsulated within that function. 

I agree that the file name is misleading. I am just changing it to patient.ts

Reply:
Larger convo, taken to [this thread](https://metriport.slack.com/archives/C04DMKE9DME/p1703807381966609).